lib/manageiq/api/common/request.rb

```diff
@@ -1,13 +1,24 @@
 module ManageIQ
   module API
     module Common
+      class RequestNotSet < ArgumentError
+        def initialize
+          super("Current request has not been set")
+        end
+      end
+
       class Request
         FORWARDABLE_HEADER_KEYS = %w(X-Request-ID x-rh-identity).freeze
 
         def self.current
           Thread.current[:current_request]
         end
 
+        def self.current!
+          raise RequestNotSet unless current
```

Comment:
What do you think of this instead?
```ruby
def self.current!
  current || raise(RequestNotSet)
end
```

Reply:
I'd love to but that's invalid syntax.  This is that specific case where you have to use `or` - `current or raise RequestNotSet`.  Would you want that?

Reply:
Turns out the parens are required syntax.